server/src/db/repositories/complianceRepository.ts

```diff
@@ -212,6 +214,28 @@ const mongoComplianceRepository: ComplianceDbRepository = {
     }
   },
 
+  async createOrSubmitForm(draftData: Partial<ComplianceReport>): Promise<string> {
+    try {
+
+      const isDraft = draftData.status == StatusEnum.DRAFT;
+      const link = isDraft ? uuidv4() : '';
+      const expirationDate = isDraft ? new Date(Date.now() + appConfig.draftExpirationTime) : undefined;
+
+      const newDraft = new Compliance({
```

Comment:
since it can also be submission, i'd rename it to newForm

Reply:
Good idea, changed.